### PR TITLE
ANY23-207 Implement Microformats2

### PR DESCRIPTION
**Pull request by lewismc**
I am initiating this PR on behalf of @nisalanirmana so we can think about closing off the GSoC work for the summer. This PR _should_ represent Microformats2 support for Apache Any23 whilst retaining compatibility for original Microformats support. 
We can discuss more on the review of the code.

### CODE REVIEW

File: core/src/main/java/org/apache/any23/extractor/html/microformats2/HItemExtractor.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
+package org.apache.any23.extractor.html.microformats2;
```
Comment (lewismc) [inline core/src/main/java/org/apache/any23/extractor/html/microformats2/HItemExtractor.java]: ALv2.0 license header required

File: core/src/main/java/org/apache/any23/extractor/html/microformats2/HItemExtractorFactory.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
+package org.apache.any23.extractor.html.microformats2;
```
Comment (lewismc) [inline core/src/main/java/org/apache/any23/extractor/html/microformats2/HItemExtractorFactory.java]: ALv2.0 license header required

File: core/src/test/java/org/apache/any23/vocab/RDFSchemaUtilsTest.java [hunk 3]
```diff
@@ -43,7 +43,7 @@
      */
     @Test
     public void testSerializeVocabulariesNTriples() {
-        serializeVocabularies(RDFFormat.NTRIPLES, 1918);
+        serializeVocabularies(RDFFormat.NTRIPLES, 2012);//1920
```
Comment (lewismc) [inline core/src/test/java/org/apache/any23/vocab/RDFSchemaUtilsTest.java]: what doe the comment //1920 represent?

File: core/src/test/java/org/apache/any23/vocab/RDFSchemaUtilsTest.java [hunk 4]
```diff
@@ -53,7 +53,7 @@ public void testSerializeVocabulariesNTriples() {
      */
     @Test
     public void testSerializeVocabulariesRDFXML() {
-        serializeVocabularies(RDFFormat.RDFXML, 4987); // Effective lines + separators.
+        serializeVocabularies(RDFFormat.RDFXML, 5252); // Effective lines + separators. //4992
```
Comment (lewismc) [inline core/src/test/java/org/apache/any23/vocab/RDFSchemaUtilsTest.java]: what doe the comment //4992 represent?

File: core/src/main/java/org/apache/any23/extractor/html/microformats2/HItemExtractor.java (hunk 1 above)
Comment (nisalanirmana) [inline core/src/main/java/org/apache/any23/extractor/html/microformats2/HItemExtractor.java]: Added :)

File: core/src/main/java/org/apache/any23/extractor/html/microformats2/HItemExtractorFactory.java (hunk 2 above)
Comment (nisalanirmana) [inline core/src/main/java/org/apache/any23/extractor/html/microformats2/HItemExtractorFactory.java]: Added :)

File: core/src/main/java/org/apache/any23/extractor/html/microformats2/HItemExtractor.java (hunk 1 above)
Comment (lewismc) [inline core/src/main/java/org/apache/any23/extractor/html/microformats2/HItemExtractor.java]: Thank you

File: core/src/main/java/org/apache/any23/extractor/html/microformats2/HItemExtractorFactory.java (hunk 2 above)
Comment (lewismc) [inline core/src/main/java/org/apache/any23/extractor/html/microformats2/HItemExtractorFactory.java]: Thank you

File: core/src/test/java/org/apache/any23/vocab/RDFSchemaUtilsTest.java (hunk 3 above)
Comment (nisalanirmana) [inline core/src/test/java/org/apache/any23/vocab/RDFSchemaUtilsTest.java]: I have now removed the comment. I was not sure about reason behind this test failure at that time, so added a comment for the future reference, later understood that this is due to changes related to added vocabularies by me.

File: core/src/test/java/org/apache/any23/vocab/RDFSchemaUtilsTest.java (hunk 4 above)
Comment (nisalanirmana) [inline core/src/test/java/org/apache/any23/vocab/RDFSchemaUtilsTest.java]: I have now removed the comment. I was not sure about reason behind this test failure at that time, so added a comment for the future reference, later understood that this is due to changes related to added vocabularies by me.

**Comment by renato2099** [comment]
This looks great @nisalanirmana !
Just two comments. Adding comments to the classes would actually help a newcomer like myself understanding better what each class does. I guess that it is almost self-explanatory with names like "HAdrExtractor" I am guessing it extracts addresses right? But that is my point, I shouldn't be guessing ;)
And the other thing, I remember seeing somewhere that all code contributed to ASF was not supposed to have the <author> tag. Maybe @lewismc  knows the truth about it.
Great work!

**Comment by lewismc** [comment]
@nisalanirmana is this PR completed now based on our discussion over on the GSoC wiki page? If it is then I can get this merged in to master branch. Thank you

**Comment by lewismc** [comment]
hey @renato2099 which tag do you mean? ta

**Comment by renato2099** [comment]
Hey @lewismc  I am talking about the @author tag, is that encouraged to be used once code has been "donated" to the ASF?

**Comment by lewismc** [comment]
Hi Renato to be honest I didn't event notice it.
I think generally speaking it probably is Not encouraged as we all share
the code. I am not fussed either way. :)

On Saturday, September 5, 2015, Renato Marroquin <notifications@github.com>
wrote:

> Hey @lewismc <https://github.com/lewismc> I am talking about the @author
> <https://github.com/author> tag, is that encouraged to be used once code
> has been "donated" to the ASF?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/any23/pull/18#issuecomment-137987959>.
>


-- 
*Lewis*

**Comment by lewismc** [comment]
OK folks I am going to go ahead and merge this issue and associate it with https://issues.apache.org/jira/browse/ANY23-207

**Pull request merged**